### PR TITLE
NN-2194 date range stats dashboard

### PR DESCRIPTION
**Pull request by JamieDraperUK**
This commit includes the following changes:-

- Integrate with new whereabouts and elite2 API's 
- Introduce date range functionality to stat page
- Introduce date range functionality to stat detail page
- Remove of redundant code
- General tidy up around tests and controllers
- Introduce a custom API get method with timeout value overrides to facilitate longish-running pages such as dashboards

Screenshot 2019-11-15 at 09 50 03

### CODE REVIEW

File: backend/utils.js [hunk 1]
```diff
@@ -38,6 +38,8 @@ const capitalize = string => {
   return lowerCase.charAt(0).toUpperCase() + lowerCase.slice(1)
 }
 
+const capitalizeStart = string => string && string[0].toUpperCase() + string.slice(1, string.length)
```
Comment (jsrobertson) [inline backend/utils.js]: Same output as `capitalize`, above?

File: backend/utils.js [hunk 2]
```diff
@@ -189,6 +176,8 @@ const stripAgencyPrefix = (location, agency) => {
   return null
 }
 
+const toUpperCamelCase = value => value.replace(/([A-Z])/g, ' $1').replace(/^./, str => str.toUpperCase())
```
Comment (jsrobertson) [inline backend/utils.js]: Is this used?

File: backend/utils.js (hunk 1 above)
Comment (jsrobertson) [reply backend/utils.js]: Is it used?

File: views/attendanceStatistics.njk [hunk 3]
```diff
@@ -3,8 +3,15 @@
 {% from "govuk/components/button/macro.njk" import govukButton %}
 {% from "govuk/components/inset-text/macro.njk" import govukInsetText %}
 {% from "components/datePicker/datePicker.njk" import datePicker %}
+{% from "govuk/components/error-summary/macro.njk" import govukErrorSummary %}
 
- {% block beforeContent %}
+{% if shouldClearFormValues  %}
```
Comment (jsrobertson) [inline views/attendanceStatistics.njk]: What is `shouldClearFormValues`?

File: backend/utils.js (hunk 1 above)
Comment (JamieDraperUK) [reply backend/utils.js]: Almost, the above lowers the whole string before uppercasing the first character.  Whereas the new one doesn't.

File: views/attendanceStatistics.njk (hunk 3 above)
Comment (JamieDraperUK) [reply views/attendanceStatistics.njk]: Boolean, see the attendanceStatistic.js changes, you'll need to expand it to see it. Some are collapsed via the diff viewer.

File: backend/utils.js (hunk 2 above)
Comment (JamieDraperUK) [reply backend/utils.js]: Good spot!

File: views/attendanceStatistics.njk (hunk 3 above)
Comment (jsrobertson) [reply views/attendanceStatistics.njk]: Doh

File: backend/controllers/attendanceStatistics.js [hunk 4]
```diff
@@ -1,126 +1,247 @@
...
+
+  return `${readableDateFormat(fromDate, 'D/MM/YYYY')}`
+}
+
+const getStatPresetsLinks = ({ activeCaseLoadId }) => {
+  const weeklyDateRange = dateRangeForStats()
+  const previousWeeklyDateRange = dateRangeForStats(1)
+  const statsFor2WeeksAgoDateRange = dateRangeForStats(2)
+
+  const statsForCurrentWeek = `/manage-prisoner-whereabouts/attendance-reason-statistics?agencyId=${activeCaseLoadId}&period=AM_PM&fromDate=${
```
Comment (jsrobertson) [inline backend/controllers/attendanceStatistics.js]: Might be neater if `/manage-prisoner-whereabouts/attendance-reason-statistics?agencyId=${activeCaseLoadId}&period=AM_PM` or at least `/manage-prisoner-whereabouts/attendance-reason-statistics` wasn't repeated.

Comment (JamieDraperUK) [reply backend/controllers/attendanceStatistics.js]: Agreed! Will sort it now.

**Pull request merged**